include/HTTPClient.h

```diff
@@ -61,16 +78,19 @@ class HTTPClient {
 		size_t chunkRemaining = 0;
 		bool isStreaming = false;
 		size_t readRaw(char *dst);
-		bool skip(size_t len, bool dontRead = false);
+		bool skipRaw(size_t len, bool dontRead = false);
 	};
 
+	typedef std::unique_ptr<struct HTTPClient::HTTPResponse> HTTPResponse_t;
+
   private:
-	static void executeImpl(const struct HTTPRequest &request, const char *url, struct HTTPResponse *&response);
+	static HTTPResponse_t executeImpl(const struct HTTPRequest &request, HTTPResponse_t response);
```

Comment:
isn't it better to pass it via reference here?

Reply:
HTTPResponse_t is actually just a unique_ptr so I think it shouldn't really be passed by reference. I added a typedef for this to make method signatures a bit shorter, but maybe naming it differently would be better as currently it can get confused with a struct typedef.